src/hooks/protections/received_event.cpp

```diff
@@ -751,7 +890,59 @@ namespace big
 			}
 			break;
 		}
-		default: break;
+		case eNetworkEvents::ACTIVATE_VEHICLE_SPECIAL_ABILITY_EVENT:
+		{
+			int16_t net_id = buffer->Read<int16_t>(13);
+
+			if (is_local_vehicle(net_id))
+			{
+				g_pointers->m_gta.m_send_event_ack(event_manager, source_player, target_player, event_index, event_handled_bitset);
+				return;
+			}
+
+			buffer->Seek(0);
+			break;
+		}
+		case eNetworkEvents::DOOR_BREAK_EVENT:
+		{
+			int16_t net_id = buffer->Read<int16_t>(13);
+
+			if (is_local_vehicle(net_id))
+			{
+				g_pointers->m_gta.m_send_event_ack(event_manager, source_player, target_player, event_index, event_handled_bitset);
+				return;
+			}
+
+			buffer->Seek(0);
+			break;
+		}
+		case eNetworkEvents::CHANGE_RADIO_STATION_EVENT:
+		{
+			int16_t net_id = buffer->Read<int16_t>(13);
+
+			if (is_local_vehicle(net_id))
+			{
+				if (!is_in_vehicle(plyr->get_ped()))
+				{
+					g_pointers->m_gta.m_send_event_ack(event_manager, source_player, target_player, event_index, event_handled_bitset);
+					return;
+				}
+
+				if (plyr->m_radio_station_change_rate_limit.process())
+				{
+					g_pointers->m_gta.m_send_event_ack(event_manager, source_player, target_player, event_index, event_handled_bitset);
+					return;
+				}
+			}
+			else
+			{
+				g_pointers->m_gta.m_send_event_ack(event_manager, source_player, target_player, event_index, event_handled_bitset);
+				return;
+			}

```

Comment:
`is_local_vehicle` has check for `g_local_player->m_vehicle->m_driver == g_local_player` so if you are passenger and driver change the radio station the `else` case will be called and event is ignored.

Reply:
so if your friend change the radio and you ignore it, thats bad isn't